src/app/parser.rs

```diff
@@ -922,7 +922,15 @@ where
                     }
 
                     if arg_os.starts_with(b"--") {
-                        needs_val_of = self.parse_long_arg(matcher, &arg_os)?;
+                        let long_arg_parse = self.parse_long_arg(matcher, &arg_os);
+                        if let Err(e) = long_arg_parse {
+                            if e.kind == ErrorKind::UnknownArgument {
+                                continue;
```

Comment:
Instead of `continue` which will start parsing the next arg, we should use something like `()` to just "do nothing". Better yet, we could even just say `if e.kind != ErrorKind::UnknownArgument { return Err(e); }` which returns the error only in the case it's not an unknown arg, but just ignores it otherwise.